workers/unity/Packages/com.improbable.gdk.core/Worker/ComponentOpDeserializer.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    internal static class ComponentOpDeserializer
+    {
+        private static readonly Dictionary<uint, IComponentDiffDeserializer> ComponentIdToComponentDeserializer =
+            new Dictionary<uint, IComponentDiffDeserializer>();
+
+        private static readonly Dictionary<(uint, uint), ICommandDiffDeserializer> CommandIdsToCommandDeserializer =
+            new Dictionary<(uint, uint), ICommandDiffDeserializer>();
+
+        static ComponentOpDeserializer()
+        {
+            foreach (var type in ReflectionUtility.GetNonAbstractTypes(typeof(IComponentDiffDeserializer)))
+            {
+                var instance = (IComponentDiffDeserializer) Activator.CreateInstance(type);
+                ComponentIdToComponentDeserializer.Add(instance.GetComponentId(), instance);
+            }
+
+            foreach (var type in ReflectionUtility.GetNonAbstractTypes(typeof(ICommandDiffDeserializer)))
+            {
+                var instance = (ICommandDiffDeserializer) Activator.CreateInstance(type);
+                CommandIdsToCommandDeserializer.Add((instance.GetComponentId(), instance.GetCommandId()), instance);
+            }
+        }
+
+        public static void DeserializeAndAddComponent(AddComponentOp op, ViewDiff viewDiff)
+        {
+            if (!ComponentIdToComponentDeserializer.TryGetValue(op.Data.ComponentId, out var deserializer))
+            {
+                throw new ArgumentException($"Can not deserialize component with ID {op.Data.ComponentId}");
+            }
+
+            deserializer.AddComponentToDiff(op, viewDiff);
+        }
+
+        public static void DeserializeAndApplyComponentUpdate(ComponentUpdateOp op, ViewDiff viewDiff, uint updateId)
+        {
+            if (!ComponentIdToComponentDeserializer.TryGetValue(op.Update.ComponentId, out var deserializer))
+            {
+                throw new ArgumentException($"Can not deserialize component with ID {op.Update.ComponentId}");
+            }
+
+            deserializer.AddUpdateToDiff(op, viewDiff, updateId);
+            ++updateId;
+        }
+
+        public static void DeserializeAndApplyCommandRequestReceived(CommandRequestOp op, ViewDiff viewDiff)
+        {
+            if (!CommandIdsToCommandDeserializer.TryGetValue(
+                (op.Request.ComponentId, op.Request.SchemaData.Value.GetCommandIndex()),
+                out var deserializer))
+            {
+                throw new ArgumentException($"Can not deserialize component with ID {op.Request.ComponentId}");
```

Comment:
Probably want to print which command as well. 

Same for the response received below.

Reply:
Sure